dialect/sql/builder.go

```diff
@@ -1604,14 +1632,29 @@ func (s *Selector) Columns(columns ...string) []string {
 	return names
 }
 
-// On sets the `ON` clause for the `JOIN` operation.
-func (s *Selector) On(c1, c2 string) *Selector {
+// Ons sets or appends the given predicate to the statement.
+func (s *Selector) Ons(p *Predicate) *Selector {
 	if len(s.joins) > 0 {
-		s.joins[len(s.joins)-1].on = fmt.Sprintf("%s = %s", c1, c2)
+		join := &s.joins[len(s.joins)-1]
+
+		switch {
+		case join.on == nil:
+			join.on = p
+		default:
+			join.on = And(join.on, p)
+		}
```

Comment:
Change it to `OnP`.

Reply:
if use `Onp`
switch case default `And` seems redundant
i think so
remove default case ?

Reply:
I meant, change the function name from `Ons` to `OnP` (i.e. "On Predicate").